sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLSessionManager.scala

```diff
@@ -88,6 +88,7 @@ private[hive] class SparkSQLSessionManager(hiveServer: HiveServer2, sqlContext:
     HiveThriftServer2.eventManager.onSessionClosed(sessionHandle.getSessionId.toString)
     val ctx = sparkSqlOperationManager.sessionToContexts.getOrDefault(sessionHandle, sqlContext)
     ctx.sparkSession.sessionState.catalog.getTempViewNames().foreach(ctx.uncacheTable)
+    ctx.sparkSession.sessionState.closeState()
```

Comment:
this seems a bug to me for single session mode when all remote client sessions share the same instance of SparkSession/SessionState.

Reply:
@yaooqinn Thanks for your review.
when `SparkSQLSessionManager.closeSession()` is called, it indicates that the sparkSession should be closed, so we should also call close `SparkSession.sessionState`.